### ISSUE TITLE
loss=nan or non decreasing loss

### ISSUE THREAD

**Issue by fariba87** [hbb1/2d-gaussian-splatting#119]
hello. thank you for your great work.
i want to do 3d reconstruction on a small object, which i captured around 40 images while it is circling on a turntable. i used pinhole camera model in colmap GUI to extract camera parameters. the images' resolution is ~4000*3000 and i  consider to use principle point in the center of image.
i have some questions
1) should i use undistorted images for training?
2) for training i use  python train.py -s mydatasetpath -r 2 --depth_ratio 1 --lambda_dist 100 --lambda_normal 100 but after some iteraration(~100) loss goes to nan. what should be the reason?
3) sometime by resizing images( 16 times smaller) loss will not be nan but it is not decreasing and just fluctuating around a number. how can i change the parameters to have decreasing loss?

**Comment by hbb1** [comment]
This is wired. Did you check if the camera poses are correct ? I have provided a colmap extractor in the convert.py for your references.

**Comment by fariba87** [comment]
> This is wired. Did you check if the camera poses are correct ? I have provided a colmap extractor in the convert.py for your reference.

yes. in convert.py camera model is set as "OPENCV" . i used GUI version of colmap to create sparse folder both with "pinhole" and "opencv" model. but still same problem.
i tried to evaluate render.py up to the point loss is not nan. but the final mesh is wrong(even though point cloud created during training seems right ]


![image](https://github.com/user-attachments/assets/73dca0fa-31ab-4627-992c-d5847aeac215)

**Comment by hbb1** [comment]
How about the rendering results?

**Comment by vivaansinghvi07** [comment]
I was having this issue due to a differing version of the `diff_surfel _rasterization` package (particularly the `glm` third party submodule I think). What fixed it was creating fresh copies of the submodules (through `git submodule update --init --recursive` on a fresh repo, not updating the submodules with `--remote` or anything) and reinstalling them
